library/src/main/kotlin/org/contentauth/c2pa/manifest/ManifestHelpers.kt

```diff
@@ -0,0 +1,430 @@
+
+        val attestationBuilder = AttestationBuilder()
+        attestationBuilder.addAssertionMetadata {
+            device(deviceName)
+            dateTime(getCurrentTimestamp())
+            addMetadata("capture_method", "screenshot")
+            appName?.let { addMetadata("source_application", it) }
+        }
+
+        attestationBuilder.buildForManifest(builder)
+        return builder
+    }
+
+    fun createSocialMediaShareManifest(
+        originalTitle: String,
+        platform: String,
+        originalFormat: String = C2PAFormats.JPEG,
+        format: String = C2PAFormats.JPEG,
+        claimGeneratorName: String = "Android C2PA SDK",
+        claimGeneratorVersion: String = "1.0.0"
+    ): ManifestBuilder {
+        val ingredient = Ingredient(
+            title = originalTitle,
+            format = originalFormat,
+            relationship = C2PARelationships.PARENT_OF
+        )
+
+        var softwareAgent = SoftwareAgent(claimGeneratorName, android.os.Build.VERSION.CODENAME, android.os.Build.VERSION.BASE_OS)
+
+
+        return ManifestBuilder()
+            .title("Shared on $platform")
+            .format(format)
+            .claimGenerator(claimGeneratorName, claimGeneratorVersion)
+            .addIngredient(ingredient)
+            .addAction(Action(
+                action = C2PAActions.RECOMPRESSED,
+                whenTimestamp = getCurrentTimestamp(),
+                softwareAgent = softwareAgent,
+                reason = "Social media optimization",
+                digitalSourceType = DigitalSourceTypes.HUMAN_EDITS
+            ))
+            .addAction(Action(
+                action = C2PAActions.PUBLISHED,
+                whenTimestamp = getCurrentTimestamp(),
+                softwareAgent = softwareAgent
+            ))
+    }
+
+    fun createFilteredImageManifest(
+        originalTitle: String,
+        filterName: String,
+        originalFormat: String = C2PAFormats.JPEG,
+        format: String = C2PAFormats.JPEG,
+        appName: String? = null,
+        claimGeneratorName: String = "Android C2PA SDK",
+        claimGeneratorVersion: String = "1.0.0"
+    ): ManifestBuilder {
+
+        var softwareAgent = SoftwareAgent(claimGeneratorName, android.os.Build.VERSION.CODENAME, android.os.Build.VERSION.BASE_OS)
+
+        val ingredient = Ingredient(
+            title = originalTitle,
+            format = originalFormat,
+            relationship = C2PARelationships.PARENT_OF
+        )
+
+        val filterParameters = mapOf(
+            "filter_name" to filterName,
+            "filter_type" to "digital_filter"
+        )
+
+        return ManifestBuilder()
+            .title("$originalTitle (Filtered)")
+            .format(format)
+            .claimGenerator(claimGeneratorName, claimGeneratorVersion)
+            .addIngredient(ingredient)
+            .addAction(Action(
+                action = C2PAActions.FILTERED,
+                whenTimestamp = getCurrentTimestamp(),
```

Comment:
I don't believe this field is defined by the spec, [see here](https://spec.c2pa.org/specifications/specifications/2.2/specs/C2PA_Specification.html#_schema_and_example_9).